commit 88e1e66a2a1a1ef221fee633e099acd2eaa7e55c
Author: Harald Freudenberger <freude@linux.ibm.com>
Date:   Wed Jan 23 13:41:35 2019 +0100

    s390/zcrypt: fix specification exception on z196 during ap probe
    
    commit 8f9aca0c45322a807a343fc32f95f2500f83b9ae upstream.
    
    The older machines don't have the QCI instruction available.
    With support for up to 256 crypto cards the probing of each
    card has been extended to check card ids from 0 up to 255.
    For machines with QCI support there is a filter limiting the
    range of probed cards. The older machines (z196 and older)
    don't have this filter and so since support for 256 cards is
    in the driver all cards are probed. However, these machines
    also require to have the card id fit into 6 bits. Exceeding
    this limit results in a specification exception which happens
    on every kernel startup even when there is no crypto configured
    and used at all.
    
    This fix limits the range of probed crypto cards to 64 if
    there is no QCI instruction available to obey to the older
    ap architecture and so fixes the specification exceptions
    on z196 machines.
    
    Cc: stable@vger.kernel.org # v4.17+
    Fixes: af4a72276d49 ("s390/zcrypt: Support up to 256 crypto adapters.")
    Signed-off-by: Harald Freudenberger <freude@linux.ibm.com>
    Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/s390/crypto/ap_bus.c b/drivers/s390/crypto/ap_bus.c
index f039266b275d..a57b969b8973 100644
--- a/drivers/s390/crypto/ap_bus.c
+++ b/drivers/s390/crypto/ap_bus.c
@@ -249,7 +249,8 @@ static inline int ap_test_config(unsigned int *field, unsigned int nr)
 static inline int ap_test_config_card_id(unsigned int id)
 {
 	if (!ap_configuration)	/* QCI not supported */
-		return 1;
+		/* only ids 0...3F may be probed */
+		return id < 0x40 ? 1 : 0;
 	return ap_test_config(ap_configuration->apm, id);
 }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1426_88e1e66a2a1a1ef221fee633e099acd2eaa7e55c.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1426_88e1e66a2a1a1ef221fee633e099acd2eaa7e55c.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
